nanofi/include/api/nanofi.h

```diff
@@ -135,16 +274,62 @@ flow_file_record* create_ff_object(const char *file, const size_t len, const uin
 
 flow_file_record* create_ff_object_na(const char *file, const size_t len, const uint64_t size);
 
-void free_flowfile(flow_file_record*);
+/**
+ * Get incoming flow file. To be used in processor logic callbacks.
+ * @param session current processor session
+ * @param context current processor context
+ * @return a flow file record or nullptr in case there is none in the session
+ **/
+flow_file_record* get_flowfile(processor_session* session, processor_context* context);
+
+
+/**
+ * Free flow file
+ * @param ff flow file
+ **/
+void free_flowfile(flow_file_record* ff);
 
+/**
+ * Adds an attribute, fails in case there is already an attribute with the given key.
+ * @param ff flow file
+ * @param key name of attribute
+ * @param value location of value
+ * @size size size of the data pointed by "value"
+ * @return 0 in case of success, -1 otherwise (already existed)
+ **/
 uint8_t add_attribute(flow_file_record*, const char *key, void *value, size_t size);
 
-void update_attribute(flow_file_record*, const char *key, void *value, size_t size);
+/**
+ * Updates an attribute (adds if it hasn't existed before)
+ * @param ff flow file
+ * @param key name of attribute
+ * @param value location of value
+ * @size size size of the data pointed by "value"
+ **/
+void update_attribute(flow_file_record* ff, const char *key, void *value, size_t size);
 
-uint8_t get_attribute(flow_file_record *ff, attribute *caller_attribute);
+/**
+ * Get the value of an attribute. Value and value size are written to parameter "caller_attribute"
+ * @param ff flow file
+ * @param caller_attribute attribute structure to provide name and get value, size
+ * @return 0 in case of success, -1 otherwise (no such attribute)
+ **/
+uint8_t get_attribute(const flow_file_record *ff, attribute *caller_attribute);
 
+/**
+ * Get the quantity of attributes
+ * @param ff flow file
+ * @return the number of attributes
+ **/
 int get_attribute_qty(const flow_file_record* ff);
```

Comment:
I thought there was a ticket for this ( can't find it so maybe it was never made ?? ) but we shouldn't be using qty or similar within an API. This should be count or quantity.